src/Engine/ProtoCore/FFI/FFIExecutionManager.cs

```diff
@@ -36,6 +37,16 @@ public Assembly LoadAssembly(string name)
             {
                 try
                 {
+                    //if the assembly is named system.private.xyz -
+                    //then we can't load it, and it's very likely
+                    //already loaded, just return it.
```

Comment:
Does it fail to load because it's already loaded?

Reply:
As per the net6 issue that you posted, maybe we should just check for the full assembly name(it is the only one that is  special)

Reply:
I guess we can revisit this if needed.
I am fine with merging as is